pandas/core/indexes/timedeltas.py

```diff
@@ -474,6 +474,10 @@ def intersection(self, other):
             return self
         if len(other) == 0:
             return other
+
+        self = self.sort_values()
```

Comment:
Sorting can be relatively expensive. Are we able to avoid this?

I think this would also break `sort=False`.

Reply:
I would think that we could extract the `i8` values and re-use `Int64Index.sort`. Is that an option?